src/main/java/net/citizensnpcs/EventListen.java

```diff
@@ -559,4 +570,45 @@ public int hashCode() {
             return prime * (prime * (prime + ((worldName == null) ? 0 : worldName.hashCode())) + x) + z;
         }
     }
+
+    private class SkinUpdateTracker {
+        float initialYaw;
+        final Location location = new Location(null, 0, 0, 0);
+        boolean hasMoved;
+        int rotationCount;
+
+        SkinUpdateTracker(Player player) {
+            reset(player);
+        }
+
+        boolean shouldUpdate(Player player) {
+
+            if (!hasMoved) {
+                hasMoved = true;
+                return true;
+            }
+
+            Location currentLoc = player.getLocation(YAW_LOCATION);
+            float currentYaw = currentLoc.getYaw();
+
+            boolean hasRotated = Math.abs(currentYaw - this.initialYaw) < 90;
```

Comment:
Oh wow, completely forgot about that.

I'll explain the turning issue. In testing, if I log in or am teleported, if I wait perhaps 5 seconds or so before turning around, NPCs behind me do not load skins. I'm sending updates after the player turns at least 90 degrees. I experimented with smaller numbers than 90 but it caused more updates and more tab list flashing and didn't seem to provide any benefit over 90.

Ill add a comment.